app/commands/document/search_docs.rb

```diff
@@ -0,0 +1,106 @@
+class Document
+  class SearchDocs
+    include Mandate
+
+    DEFAULT_PAGE = 1
+    DEFAULT_PER = 25
+
+    def self.default_per
+      DEFAULT_PER
+    end
+
+    def initialize(criteria: nil, track_slug: nil, page: nil, per: nil)
+      @criteria = criteria
+      @track_slug = track_slug
+      @page = page.present? && page.to_i.positive? ? page.to_i : DEFAULT_PAGE
+      @per = per.present? && per.to_i.positive? ? per.to_i : self.class.default_per
+    end
+
+    def call
+      results = Exercism.opensearch_client.search(index: Document::OPENSEARCH_INDEX, body: search_body)
+
+      doc_ids = results["hits"]["hits"].map { |hit| hit["_source"]["id"] }
+      docs = doc_ids.present? ?
+        Document.where(id: doc_ids).
+          includes(:track).
+          order(Arel.sql("FIND_IN_SET(id, '#{doc_ids.join(',')}')")).
+          to_a : []
+
+      total_count = results["hits"]["total"]["value"].to_i
+      Kaminari.paginate_array(docs, total_count: total_count).
+        page(page).per(per)
+    rescue StandardError => e
+      Bugsnag.notify(e)
+      Fallback.(criteria, track_slug, page, per)
+    end
+
+    private
+    attr_reader :criteria, :track_slug, :page, :per, :docs
+
+    def search_body
+      {
+        query: search_query,
+        sort: :_score,
+
+        # Only return the solution IDs, not the entire document, to improve performance
+        _source: [:id],
+
+        # Paging information
+        from: (page - 1) * per,
+        size: per
+      }
+    end
+
+    def search_query
+      {
+        bool: {
+          must: [
+            track_slug.blank? ? nil : { terms: { 'track.slug': [track_slug].flatten } },
+            criteria.blank? ? nil : {
+              query_string: {
```

Comment:
@iHiD The `query_string` option allows us to do wildcard searches (`*XYZ*`). While flexible, this is less performant than the default query search, which searches words. I don't think this is an issue for the documents, but it is something to consider to the solution searching.